doc/user/tutorials/making_your_first_app.rst

```diff
@@ -0,0 +1,510 @@
+
+    from myapp.plugin import DummyPhysicsPlugin
+
+    class ExampleApp(App):
+        def __init__(self):
+            # activate all built-in plugins
+            App.__init__(self)
+
+            # register our plugin with the plugin manager
+            self._pm.register(DummyPhysicsPlugin) 
+
+        @property
+        def splashText(self):
+            return "** My Example App **"
+
+
+Defining the physics plugin
+===========================
+Now we will create the plugin that will coordinate our dummy physics modules. 
+
+.. admonition:: What are plugins again? 
+
+    Plugins are the basic modular building block of ARMI-based apps. In some cases, one plugin
+    will be associated with one physics kernel (like MCNP). This is a reasonable practice when
+    you expect to be mixing and matching various combinations of plugins between related
+    teams. It is also possible to have a plugin that performs a whole cacophony of analyses
+    using multiple codes, which some smaller research teams may find preferable. The
+    flexibility is very broad.
+
+    See :py:mod:`armi.plugins` more for info.
+
+Plugin code can exist in any directory structure in an app. In this very simple app we
+simply put it in the :file:`myapp/plugins.py` file.
+
+.. note:: For "serious" plugins, we recommend mirroring the ``armi/physics/[subphysics]``
+    structure of the ARMI framework physics plugin folder.
+
+We will start the plugin simply by pointing to the two physics kernels we wish to
+register. We hook them in and tell ARMI the ``ORDER`` they should be run in based on
+the built-in ``STACK_ORDER`` attribute. We will come back to this plugin definition later
+on to do add a little more to the plugin. 
+
+
+.. code-block:: python
+    :caption: ``myapp/plugin.py``
+
+    from armi import plugins
+    from armi import interfaces
+    from armi.interfaces import STACK_ORDER as ORDER
+
+    from myapp import fluxSolver
+    from myapp import thermalSolver
+
+
+    class DummyPhysicsPlugin(plugins.ArmiPlugin):
+        @staticmethod
+        @plugins.HOOKIMPL
+        def exposeInterfaces(cs):
+            kernels = [
+                interfaces.InterfaceInfo(ORDER.FLUX, fluxSolver.FluxInterface, {}),
+                interfaces.InterfaceInfo(ORDER.THERMAL_HYDRAULICS, thermalSolver.ThermalInterface, {}),
+            ]
+            return kernels
+
+Creating the physics kernels
+============================
+So far we have basically been weaving an administrative thread to tell ARMI about the code
+we want to run. Now we finally get to write the guts of the code that actually does
+something. In your real app, this code will run your own industrial or research code, or
+perform your own methodology.  Here we just have it make up dummy values representing flux
+and temperatures.
+
+Making the dummy flux kernel
+----------------------------
+In a previous tutorial, we made a function that sets a dummy flux to all parts of the core
```

Comment:
Link to the previous tutorial just in case they missed it or want to go back and find it to remind themselves of the tutorial.